src/Stratis.Bitcoin/P2P/PeerAddress.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.Net;
+using NBitcoin.Protocol;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Stratis.Bitcoin.P2P
+{
+    [JsonObject]
+    public sealed class PeerAddress
+    {
+        private const int PeerAddressLastSeen = 30;
+        private const int PeerMinimumFailDays = 7;
+        private const int PeerMaximumWeeklyAttempts = 10;
+        private const int PeerMaximumConnectionRetries = 3;
+
+        #region Address Data
+
+        [JsonProperty]
+        private IPEndPoint endPoint;
+
+        [JsonProperty]
+        private DateTimeOffset? addressTime;
+
+        [JsonIgnore]
+        public NetworkAddress NetworkAddress
+        {
+            get
+            {
+                if (this.endPoint == null)
+                    return null;
+
+                var networkAddress = new NetworkAddress(this.endPoint);
+                if (this.addressTime != null)
+                    networkAddress.Time = this.addressTime.Value;
+
+                return networkAddress;
+            }
+        }
+
+        [JsonProperty]
+        private string loopback;
+        [JsonIgnore]
+        public IPAddress Loopback
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(this.loopback))
+                    return null;
+                return IPAddress.Parse(this.loopback);
+            }
+        }
+
+        #endregion
+
+        #region Connection Data
+
+        [JsonProperty]
+        public int ConnectionAttempts { get; private set; }
+        [JsonProperty]
+        public DateTimeOffset? LastConnectionHandshake { get; private set; }
+
+        [JsonIgnore]
+        public bool IsNew
+        {
+            get
+            {
+                return
+                    this.LastConnectionAttempt == null &&
+                    this.LastConnectionSuccess == null &&
+                    this.LastConnectionHandshake == null;
+            }
+        }
+
+        [JsonProperty]
+        public DateTimeOffset? LastConnectionAttempt { get; private set; }
+        [JsonProperty]
```

Comment:
Space.

We also need comments but I am fine it will be in the next PR.

Reply:
Fixed.

Reply:
Will add comments.